File: clients/java/examples/collections-date.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.Update;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Filters;
import java.time.Instant;
import java.util.Calendar;
import java.util.Date;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    Calendar calendar = Calendar.getInstance();

    // Use date in insertions
    collection.insertOne(new Document().append("registered_at", calendar));
    collection.insertOne(new Document().append("date_of_birth", new Date()));
    collection.insertOne(new Document().append("just_a_date", Instant.now()));

    // Use date in a filter
    collection.updateOne(
        Filters.eq("registered_at", calendar), Update.create().set("message", "happy Sunday!"));
    collection.findOne(
        Filters.lt("date_of_birth", new Date(System.currentTimeMillis() - 1000 * 1000)));
  }
}

File: clients/csharp/examples/collections-date.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Use date in an insertion
    var document = new Document()
    {
      { "registered_at", new DateTime(2020, 1, 1, 1, 1, 0) },
      { "hire_date", new DateOnly(2020, 1, 1) },
      { "shift_start", new TimeOnly(9, 15) },
    };

    await collection.InsertOneAsync(document);

    // Use date in a filter
    var filterBuilder = Builders<Document>.CollectionFilter;
    var filter = filterBuilder.And(
      filterBuilder.Eq(
        "registered_at",
        new DateTime(2020, 1, 1, 1, 1, 0)
      ),
      filterBuilder.Lt("hire_date", new DateOnly(2022, 11, 4))
    );

    await collection.FindOneAsync(filter);

    // Use date in an update
    var update = Builders<Document>
      .CollectionUpdate.Set("shift_start", new TimeOnly(10, 30))
      .Set("registered_at", new DateTime(2020, 3, 5, 1, 1, 0));

    await collection.UpdateOneAsync(filter, update);
  }
}
